### PR TITLE
SAK-47084 Discussions: Dates are on bad format when trying saving a new forum without title on "Specify dates to open (show) and/or close (hide)" option

### PR DESCRIPTION
**Pull request by feralvarez95**
https://sakaiproject.atlassian.net/browse/SAK-47084

- Toggle that displays date selectors was not working fine in some situations.
- Dates were lost or bad formatted if no title was inserted.

### CODE REVIEW

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionForumBean.java [hunk 1]
```diff
@@ -533,16 +535,14 @@ public String getOpenDate(){
...
-			StringBuilder dateTimeOpenDate = new StringBuilder( datetimeFormat.format( forum.getOpenDate() ) );			
-			return dateTimeOpenDate.toString();
+			return userTimeService.dateTimeFormat(forum.getOpenDate().toInstant(), FormatStyle.SHORT, null);
 		}
-	}	  
+	}
 
 	public void setOpenDate(String openDateStr){
-		if (StringUtils.isNotBlank(openDateStr)) {
+		if (StringUtils.isNotBlank(openDateStr) && StringUtils.isNotBlank(openDateISO)) {
```
Comment (mpellicer) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionForumBean.java:543]: StringUtils.isNoneBlank makes more sense here

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionForumBean.java [hunk 2]
```diff
@@ -556,25 +556,23 @@ public String getCloseDate(){
...
-			StringBuilder dateTimeCloseDate = new StringBuilder( datetimeFormat.format( forum.getCloseDate() ) );
-			return dateTimeCloseDate.toString();
+			return userTimeService.dateTimeFormat(forum.getCloseDate().toInstant(), FormatStyle.SHORT, null);
 		}
-	}	  
+	}
 
 	public void setCloseDate(String closeDateStr){
-		if (StringUtils.isNotBlank(closeDateStr)) {
+		if (StringUtils.isNotBlank(closeDateStr) && StringUtils.isNotBlank(closeDateISO)) {
```
Comment (mpellicer) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionForumBean.java:564]: StringUtils.isNoneBlank makes more sense here

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
Just two minor improvements.

I don't like using the jQuery datepicker anymore and we're currently replacing this by a datetime-local input, I believe we must create a webcomponent for this but that will make backports more complex.

Thanks for the good work @feralvarez95

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionForumBean.java (hunk 1 above)
Comment (feralvarez95) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionForumBean.java]: much better! Thanks

**Pull request merged**